tbtacl/tbtacl.in

```diff
@@ -114,24 +114,24 @@ esac
 
 case "$action" in
 	# New device attached, go to authorize it
-	add)	authorize $device
+	add)	authorize "$device"
 		;;
 
 	# The device got authorized, let's try to authorize again the devices
 	# behind it
-	change)	list=$device/*/authorized
+	change)	list="$device/*/authorized"  # this glob is expanded later
```

Comment:
later == line 124?

Reply:
Line 126, in `for i in $list ; do`

Reply:
It already was not being expanded on line 122 btw - globs and brace expansions are literals in assignments.  Adding quotes just makes it more explicit to the reader.

Reply:
It must be expanded in line 124 too, otherwise it'll fail. Does it?
(I have no way to test it right now. Did you?)

Reply:
Line 124 just needs to check if the `*` character appears in the $list string, so it should still be OK.  Probably could drop the `-F` from grep as well - or I could remove the quotes around $list here if you prefer.

It does seem to work this way on my system.

Reply:
oh crap - I was completely incorrect in my last comment. 124 does need to expand the glob in order for the test to work.  Fixing.

Reply:
(fixed)